spec/lib/traject/psulib_config_spec.rb

```diff
@@ -94,4 +95,27 @@
       expect(result['collection_facet']).to contain_exactly 'Arthur O. Lewis Utopia Collection.'
     end
   end
+
+  describe 'HathiTrust access' do
+    it 'returns nil when the access field is blank' do
+      id = { '001' => '1' }
+      result = @indexer.map_record(MARC::Record.new_from_hash('fields' => [id], 'leader' => leader))
+
+      expect(result['ht_access_ss']).to be_nil
+    end
+
+    it 'returns "allow" when duplicates are all mono and the access level is mixed' do
+      id = { '001' => '1000065' }
+      result = @indexer.map_record(MARC::Record.new_from_hash('fields' => [id], 'leader' => leader))
+
+      expect(result['ht_access_ss']).to contain_exactly 'allow'
+    end
+
+    it 'returns "allow" when duplicates are both mono and multi/serial and the access level is mixed' do
+      id = { '001' => '1099502' }
+      result = @indexer.map_record(MARC::Record.new_from_hash('fields' => [id], 'leader' => leader))
+
+      expect(result['ht_access_ss']).to contain_exactly 'allow'
+    end
+  end
```

Comment:
we do not need any specs for deny case?

Reply:
I'll add some.

Reply:
Counting all possible combinations of duplicate, no duplicates, allows, denies and mixes there are around 20 possible scenarios I think. For example here are the possible options for just mono (not multis, not mono/multi mixes):

1. Mono, no dupes: allow
1. Mono, no dupes: deny
1. Mono, no dupes: blank
1. Mono, dupes: deny
1. Mono, dupes: allow
1. Mono, dupes: blank
1. Mono, dupes: mix

Thoughts? Just trying to figure out which should be covered.

Reply:
I think I'll just write notes for "similar scenarios" on things like "mono" with one access level because it's the same as multi with one access or dupes that are all the same level of access 🤔

Reply:
should be in better shape now. I think it was definitely a problem that a dupe multi with mix access wasn't covered. thanks